Source/WebInspectorUI/UserInterface/Views/InlineSwatch.js

```diff
@@ -194,22 +197,30 @@ WI.InlineSwatch = class InlineSwatch extends WI.Object
         let value = this.value;
 
         switch (this._type) {
-        case WI.InlineSwatch.Type.Color:
+        case WI.InlineSwatch.Type.Color: {
+            let title = this._tooltip;
             if (!this._readOnly) {
-                let title = WI.UIString("Click to select a color.");
+                title = WI.UIString("Click to select a color.");
                 if (this._allowChangingColorFormats())
                     title += "\n" + WI.UIString("Shift-click to switch color formats.");
                 if (InspectorFrontendHost.canPickColorFromScreen())
                     title += "\n" + WI.UIString("Option-click to pick color from screen.");
-                
-                this._swatchElement.title = title;
             }
+            this._swatchElement.title = title;
             // fallthrough
-
+        }
         case WI.InlineSwatch.Type.Gradient:
             this._swatchInnerElement.style.background = value ? value.toString() : null;
             break;
 
+        case WI.InlineSwatch.Type.Variable: {
+            if (!this._variableType)
+                break;
+
+            let colorVariable = this._delegate?.inlineSwatchColorVariableCompute(this, value.toString());
```

Comment:
i think `inlineSwatchGetMatchingColorVariable` would be a better name

also this should be `this._delegate?.inlineSwatchGetMatchingColorVariable?.(this, value.toString()) || null` as there's no guarantee that `inlineSwatchColorForVariable` exists on `delegate`